pkg/migrate/attributemigration.go

```diff
@@ -0,0 +1,299 @@
+package migrate
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"sync"
+
+	"github.com/Axway/agent-sdk/pkg/api"
+	v1 "github.com/Axway/agent-sdk/pkg/apic/apiserver/models/api/v1"
+	mv1a "github.com/Axway/agent-sdk/pkg/apic/apiserver/models/management/v1alpha1"
+	defs "github.com/Axway/agent-sdk/pkg/apic/definitions"
+	"github.com/Axway/agent-sdk/pkg/config"
+	"github.com/Axway/agent-sdk/pkg/util"
+)
+
+const queryByRefName = "metadata.references.name"
+
+var oldAttrs = []string{
+	defs.AttrPreviousAPIServiceRevisionID,
+	defs.AttrExternalAPIID,
+	defs.AttrExternalAPIPrimaryKey,
+	defs.AttrExternalAPIName,
+	defs.AttrExternalAPIStage,
+	defs.AttrCreatedBy,
+}
+
+// AttrMigrator interface for performing an attribute migration
+type AttrMigrator interface {
+	Migrate(ri *v1.ResourceInstance) (*v1.ResourceInstance, error)
+}
+
+var regexes = make([]string, 0)
```

Comment:
Do we still want to allow regexes in this? It could be dangerous to allow an agent to set that and we move more than necessary.

Reply:
I think we need it. The AWS discovery agent has this line
```
serviceBody.Attributes[serviceBody.Stage+"Hash"] = convertUnitToString(serviceBodyHash)
```

Since the attribute is not a constant I figured we needed a regex to match this one.

Reply:
I forgot about that one, good call